File: code/backend/php/src/Adapter/Rest/EchoController.php
<?php

namespace Games\Paths\Adapter\Rest;

use Games\Paths\Core\Port\EchoPort;
use Psr\Http\Message\ResponseInterface as Response;
use Psr\Http\Message\ServerRequestInterface as Request;

class EchoController
{
    private EchoPort $echoService;

    public function __construct(EchoPort $echoService)
    {
        $this->echoService = $echoService;
    }

    public function getStatus(Request $request, Response $response, array $args): Response
    {
        $status = $this->echoService->getStatus();
        
        $response->getBody()->write(json_encode($status));
        return $response->withHeader('Content-Type', 'application/json')->withStatus(200);
    }
}

File: code/backend/java/adapter-rest/src/main/java/games/paths/adapters/rest/controller/EchoController.java
package games.paths.adapters.rest.controller;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import games.paths.core.port.EchoPort;

import java.util.LinkedHashMap;
import java.util.Map;

/**
 * EchoController - REST adapter that exposes the EchoPort.
 * GET /api/echo/status → returns timestamp, server status, and server properties.
 */
@RestController
@RequestMapping("/api/echo")
public class EchoController {

    private final EchoPort echoPort;

    public EchoController(EchoPort echoPort) {
        this.echoPort = echoPort;
    }

    @GetMapping("/status")
    public Map<String, Object> getStatus() {
        Map<String, Object> response = new LinkedHashMap<>();
        response.put("status", echoPort.getServerStatus());
        response.put("timestamp", echoPort.getTimestamp());
        response.put("properties", echoPort.getServerProperties());
        return response;
    }
}
